public/app/features/scenes/core/SceneObjectBase.tsx

```diff
@@ -98,19 +103,32 @@ export abstract class SceneObjectBase<TState extends SceneObjectState = {}> impl
 
   activate() {
     this.isActive = true;
-
-    const { $data } = this.state;
-    if ($data && !$data.isActive) {
-      $data.activate();
+    if (Object.prototype.hasOwnProperty.call(this.state, '$data')) {
+      const dataObj = (this.state as any).$data;
+      if (dataObj instanceof SceneDataObject) {
+        if (!dataObj.isActive) {
+          // Data-providing objects are separate from the layout hierarchy. But they need a way for resolving scoped context,
+          // hence they reuse its consumer context
+          dataObj.setCtxResolver(this.getContext);
+          dataObj.activate();
+        }
+        dataObj._refCounter.add(this.state.key!);
+      }
```

Comment:
I really struggle to understand what we gain from adding this complexity. This seems to just complicate things

Reply:
This will set the context resolver to whatever node happens to be activated first, which feels like it could cause issues (Say given lazy rendering, and different raise conditions). 

If we have a need for one part of the object graph to share data with a different part of the tree I was thinking of using a special SceneDataRef scene object that can subscribe to data and set it in its own state.